### ISSUE TITLE
Sonarqube 9.x support

### ISSUE THREAD

**Issue by jchable** [cnescatlab/sonar-cnes-report#270]
#### Is your feature request related to a problem? Please describe.

> Yes, Sonar 9.x is not supported for report generation.
Detected SonarQube version: 9.0.1
[ERROR] SonarQube instance is not supported by cnesreport.

#### Describe the solution you'd like

> Add support for Sonar 9.x

#### Describe alternatives you've considered

> Nope :(

**Comment by Sancretor** [comment]
Hi @jchable 

Thanks for your feedback !
Unfortunately, our last official release only supports SonarQube 7.9 LTS as stated in the README, and our 8.9 LTS version is soon to be released.
As our small team is maintaining this tool during its free time, we only work on SonarQube LTS versions.

Anyway, if you'd like to help us work on support for support 9.x, Pull Requests are welcomed !

**Comment by richardcurteis** [comment]
@Sancretor any idea when the 8.9 LTS version will be available?

**Comment by Sancretor** [comment]
@richardcurteis
There's only one feature left to merge, maybe some bugfixes. 
I can't say for sure but hopefully by the end of the month. 
That's not a promise though...

**Comment by richardcurteis** [comment]
Ok great, good to know, thanks. Understand it's not certain

**Comment by jchable** [comment]
> Hi @jchable
> 
> Thanks for your feedback !
> Unfortunately, our last official release only supports SonarQube 7.9 LTS as stated in the README, and our 8.9 LTS version is soon to be released.
> As our small team is maintaining this tool during its free time, we only work on SonarQube LTS versions.
> 
> Anyway, if you'd like to help us work on support for support 9.x, Pull Requests are welcomed !

Thanks for your time and passion, I'll try to make a pull request in the next upcoming weeks for 9.x suport even if I'm not confident with sonar plugin development. We'll make our best.

**Comment by Sancretor** [comment]
Hi @jchable 
Thanks for your understanding !
We are not forcing anyone to create Pull Requests though :)
That was just a reminder that any help is welcomed.

**Comment by marcelstoer** [comment]
> @Sancretor any idea when the 8.9 LTS version will be available?

Just today I was testing the dev branch which contains the upcoming 4.0 release (see PR #279). Looking good.

**Comment by Sancretor** [comment]
And it has just been released !

**Comment by jchable** [comment]
@Sancretor Thank you and to all dev team members for it ! Keep up the good work !

**Comment by 0xERR0R** [comment]
This issue waa closed, but SorarQube 9.x is still not supported, because it is no lts, right?

**Comment by Sancretor** [comment]
@jchable Thanks a lot for the support !

@0xERR0R 
I don't know why it was closed...
And yes, we're only working for LTS compatibility as we rely on SQ APIs that will evolve through the various 9.x versions.
We don't have time to update the plugin that regularly.

**Comment by 0xERR0R** [comment]
@Sancretor OK, I understand. I tried it with SonarQube 9.1 (after changing the supported version constants in code) and it worked for me. Would it be possible to introduce a new flag "--experimental" or something like this, to disable the version check? So in this case the user can try the application even if the version is not official supported.

**Comment by Sancretor** [comment]
@0xERR0R 
I guess the initial reason we introduced the SQ version check was to avoid the huge amount of issues opened on non-compatible versions...
We may consider adding kind of a "--experimental" flag that could be useful, but this would only work for the standalone mode.
I'll create another issue because despite the official compatibility matrix, it may still (mostly) work for SQ 9.x.

In the meantime, you could still consider modifying the code and compile it yourself as a workaround.

**Comment by attahameed** [comment]
> @Sancretor OK, I understand. I tried it with SonarQube 9.1 (after changing the supported version constants in code) and it worked for me. Would it be possible to introduce a new flag "--experimental" or something like this, to disable the version check? So in this case the user can try the application even if the version is not official supported.

i am using SQ 9.2, and want to try the version that you compiled for you for v9.1. Would it be possible for you to share the jar file show how?

**Comment by ermalbaj** [comment]
so any information if this does support version 9.3 released during January 2022.
So does this plugin works with the latest version of SonarQube?
Thanks

**Comment by Sancretor** [comment]
Hi @ermalbaj 
It does not, and will not until the release of the next SonarQube 9.x LTS version.
Our plan is still unchanged and will probable remain.

**Comment by Sancretor** [comment]
Just to let you know that we just merged #314 in 'dev' branch.
Though we still don't provide official compatibility with SQ 9.x, the fix doesn't change the 8.9.x compatibility, and seems to make the plugin work on SonarQube 9.3.

**Comment by rcfja** [comment]
Latest version seems to be working on 9.4 for me. FYI to any future people you just have to create the /opt/sonarqube/.cnesreport/log directory first.

**Comment by cbarina** [comment]
@rcfja I got this to work via command line after creating that directory (thanks!) - did you manage directly in the UI?

**Comment by Sancretor** [comment]
Hi !
A quick message to let you know that SonarQube Project Manager contacted us regarding their marketplace.
Some of their paid SonarQube versions now have functionalities similar to our plugin, so we won't be authorized to publish it on the marketplace anymore, starting with SonarQube 9.5.
Of course, the plugin will still be available here, on GitHub...

**Comment by TheRealAyCe** [comment]
SonarQube 9.6.1 seems to work, although some of SonarQube's translation texts have become broken after adding the plugin JAR.

**Comment by obriat** [comment]
The /extension/cnesreport/report page works fine, but I get a 500 error when I try to generate a report (GET api/cnesreport/report?key=...)
`#web.log
Caused by: java.lang.NoClassDefFoundError: Could not initialize class fr.cnes.sonar.report.ReportCommandLine`

**Comment by obriat** [comment]
Disable and enable again the plugins fix the  500 error

**Comment by carloreggiani** [comment]
We are using the last jar (4.1.3) on Sonarqube 9.8, it seems working with few warning... any roadmap to "official" support?

**Comment by obriat** [comment]
I try to compile it locally: didn't work.
But the very first error message in the log complains about `access denied ("java.io.FilePermission" "/opt/sonarqube-9.7.1.62043/.cnesreport/log" "write")` 
So I created it manually, and it works! 
I don't know why the plugin could not create it by his own... (the sonar folder had the correct permissions)
